open-craft/frontend-app-library-authoring
# [BD-14] [SE-2947] Add license management tools.

This pull request allows library admins to set the license of their creation-- either 'All Rights Reserved' or one of the latest Creative Commons licenses. The backend PR for this change is here: https://github.com/edx/edx-platform/pull/25007

**Discussions**: https://docs.google.com/document/d/1hRrSmWWlGHhi-bh9AIlpSuLtJFnn2GpavNe8OQxthbE/edit#heading=h.x4hzuvwb0us

**Dependencies**: https://github.com/edx/frontend-app-library-authoring/pull/8

![Screen Shot 2020-09-17 at 11 24 50 AM](https://user-images.githubusercontent.com/1099483/93499445-9f739d00-f8d8-11ea-80e0-532f2a724b08.png)
![Screen Shot 2020-09-17 at 11 25 03 AM](https://user-images.githubusercontent.com/1099483/93499454-a3072400-f8d8-11ea-85da-a1aa2eab23a4.png)
![Screen Shot 2020-09-17 at 11 26 00 AM](https://user-images.githubusercontent.com/1099483/93499460-a5697e00-f8d8-11ea-8455-d4dd3297ae20.png)


**Merge deadline**: This is the review copy of the PR and is not to be merged.

**Testing instructions**:

1. Create a new library.
    1. Verify that the license field disappears if creating a legacy type library.
    2. Verify that clicking Creative Commons gives you more options for your license.
    3. Verify that the license field remembers your options if you switch from complex/video/problem to legacy and back.
    4. Verify that creating the library results in the option being saved by visiting the newly created library's edit page.
2. Edit an existing library.
    1. Verify that you can switch libraries from one license to another and that this change is saved and persists across reloads.
    2. Verify that the mutually exclusive flags 'Share Alike' and 'No Derivatives' cannot be co-selected.
3. Verify that the library's detail page shows the current license near the publish button.

**Reviewers**
